src/backend.rs

```diff
@@ -643,3 +747,21 @@ enum Expression {
         arguments: Vec<Expression>,
     },
 }
+
+impl Expression {
+    unsafe fn codegen(&self) -> *const ffi::Expression {
+        match *self {
+            Expression::Integer(value) => unsafe { ffi::create_integer(value) },
+            Expression::Variable(storage, index) => unsafe { ffi::create_integer(0) },
+            Expression::App {
+                ref function,
+                ref arguments,
+            } => match arguments.len() {
+                0 => unsafe { ffi::create_app(function.codegen(), 0) },
+                1 => unsafe { ffi::create_app(function.codegen(), 1, arguments[0].codegen()) },
+                _ => todo!(),
+            },
```

Comment:
The number of arguments passed to `created_app` is static while running JIT-ed code, but dynamic here. Do we need another function?